diff --git a/lib/typescript/index.d.ts b/lib/typescript/index.d.ts
index ffe1cad6025b04769ecc9d2831e5600a6cd3cf5e..680dfc297cea1bcc7cb8a942afa34e88720c3d38 100644
--- a/lib/typescript/index.d.ts
+++ b/lib/typescript/index.d.ts
@@ -1,4 +1,4 @@
-export { HighlightableElementProvider } from "./context";
+export { HighlightableElementProvider, useHighlightableElements } from "./context";
 export { default as HighlightableElement } from "./HighlightableElement";
 export { default as HighlightOverlay } from "./HighlightOverlay";
 export { default as FadeDuringHighlight } from "./FadeDuringHighlight";
diff --git a/src/index.ts b/src/index.ts
index e9e05f4a5cf87ae6da23476e37e7a50ef7872d1a..aed700db284f6682b0799144ddb7e0a3157da438 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,4 @@
-export { HighlightableElementProvider } from "./context";
+export { HighlightableElementProvider, useHighlightableElements } from "./context";
 export { default as HighlightableElement } from "./HighlightableElement";
 export { default as HighlightOverlay } from "./HighlightOverlay";
 export { default as FadeDuringHighlight } from "./FadeDuringHighlight";
